lib/cocoapods/generator/module_map.rb

```diff
@@ -35,14 +40,24 @@ def save_as(path)
       # @return [String]
       #
       def generate
-        <<-eos.strip_heredoc
+        result = <<-eos.strip_heredoc
           framework module #{target.product_module_name} {
             umbrella header "#{target.umbrella_header_path.basename}"
 
             export *
             module * { export * }
-          }
         eos
+
+        result << "\n#{generate_private_header_exports}" unless private_headers.empty?
+        result << "}\n"
+      end
+
+      private
+
+      def generate_private_header_exports
+        private_headers.reduce('') do |string, header|
+          string << %(  private header "#{header}\n")
+        end
```

Comment:
I'm not entirely sure if this is correct.

Shouldn't the output be something like:

```c
module Foo {
  header "Foo.h"

  explicit module Private {
    header "Foo_Private.h"
  }
}
```